### PR TITLE
feat(docker): Enable BuildKit

### PR DESCRIPTION
**Pull request by amartsinovich-dialpad**
This seem to work, however some stages failed, I'm gonna investigate if this is related https://app.circleci.com/pipelines/github/talkiq/realtime/1827/workflows/c698fe67-f2f1-459e-9c3e-327f256d44d0/jobs/8739

### CODE REVIEW

**Comment by amartsinovich-dialpad** [comment]
This PR won't affect `realtime-nlp` but I made tests and buildkit build took 34 minutes instead of 46 minutes of reference build. Not sure if this is indeed due to buildkit or just some cache tho.

File: gcloud/orb.yaml [hunk 1]
```diff
@@ -350,7 +351,7 @@ jobs:
           Name of dockerfile to use.
         type: string
       executor:
-        default: docker:18.06.3-ce-git
+        default: docker:18.09.9-git
```
Comment (TheKevJames) [inline gcloud/orb.yaml:354]: Should this match the `18.09.3` you're requesting above? May not matter, I suppose?

File: docker/orb.yaml [hunk 2]
```diff
@@ -44,8 +44,9 @@ commands:
       - run:
           name: docker build
           command: |
-            docker build \
+            DOCKER_BUILDKIT=1 docker build \
```
Comment (TheKevJames) [inline docker/orb.yaml:47]: You can also [use the `environment` field to set this](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-step). I genuinely have no idea what the difference would be, but my assumption would be if they're providing a specific method for this it's probably worth using?

File: gcloud/orb.yaml (hunk 1 above)
Comment (amartsinovich-dialpad) [reply gcloud/orb.yaml:354]: Strangely, the "real" version seems to be the one under `setup_remote_docker`. But yeah, it's nice to have them aligned

Comment (TheKevJames) [reply gcloud/orb.yaml]: So if I understand correctly, the executor would define the image you are running commands within -- in other words, the version of the docker _client_ you're using. The `setup_remote_docker` command would be for configuring the _server_. Theoretically, clients should be backwards compatible with virtually all servers, but I'm not sure that's an explicit guarantee (especially now that they've switched from semver to calver.. :().

**Pull request merged**